packages/core/src/base/staticClient.ts

```diff
@@ -21,15 +21,15 @@ export class ResponseError extends Error {
 
 /**
  * Extracts the dto object from the given response object
- * 
+ *
  * @param response the response object containing the dto
- * 
+ *
  * @returns the dto wrapped in a promise
  */
 export async function getObjectFromResponse(
   response: Response
 ): Promise<object> {
-  if (response.ok) {
+  if (response.ok || response.status === 304) {
```

Comment:
Kannan and I discussed a few things that should be captured in comments here:
1. make-fetch-happen will return cached body with status 304 for not modified
2. response.ok is 2xx only
3. Other redirection is on by default for make-fetch-happen, so other 3xx codes can be regarded as errors if they bubble up this far